File: workshop/csharp/test/CheckoutCS.Tests.Unit/Helpers/CartBuilder.cs
using System;
using System.Collections.Generic;

namespace CheckoutCS.Tests.Helpers
{
    internal class CartBuilder
    {
        private List<CartItem> _cartItems = new List<CartItem>();

        public Cart Build()
        {
            var cart =  new Cart(_cartItems);
            return cart;
        }

        public static implicit operator Cart(CartBuilder builder) => builder.Build();

        internal CartBuilder Containing(params CartItemBuilder[] cartItemBuilders)
        {
            foreach (CartItem builder in cartItemBuilders)
            {
                _cartItems.Add(builder);
            }
            return this;
        }
    }
}

File: workshop/php/tests/Helpers/CartBuilder.php
<?php
namespace CheckoutPHP\Tests\Helpers;

use CheckoutPHP\Cart;
use CheckoutPHP\CartItem;

class CartBuilder {
    /**
     * @var CartItem[]
     */
    private $_items = [];

    public function build() : Cart {
        return new Cart($this->_items);
    }

    public function __invoke() {
        return $this->build();
    }

    public function Containing(CartItemBuilder $cartItemBuilder) {
        $this->_items[] = $cartItemBuilder();
        return $this;
    }
}